lib/internal/bootstrap_node.js

```diff
@@ -196,7 +200,12 @@
       enumerable: false,
       configurable: true
     });
-    global.process = process;
+    Object.defineProperty(global, 'process', {
```

Comment:
basically, if we set this precedent I would be much more comfortable locking down *all* things that core could depend on, either by adding them to a closure ala https://github.com/bmeck/node/tree/no-globals or by freezing them on the global. There are *many* of these globals that break `npm` modules in the wild if they are overwritten. I do not see process as exceptionally high use vs `Error` for example. I am against freezing them on the global for polyfill reasons.